### ISSUE TITLE
GET /application/o/authorize/ - Errors out when using wildcard URI

### ISSUE THREAD

**Issue by Hacksawfred3232** [goauthentik/authentik#2969]
**Describe the bug**
When a OAuth2 app uses `*` (the wildcard URI), it errors out when opening up the authentication page.

**To Reproduce**
1. Setup a application that uses OAuth2 authentication and has `*` as a valid redirect URI
2. Attempt to trigger a login using that OAuth2 application.

**Expected behavior**
Should go to login page and/or consent page.

**Mitigation**
The error given by Authentik suggests that Python appears to be treating the URI as regex. Doing `.*` instead of `*` seems to fix this issue. However, this conflicts with the documentation given. As such, this is considered a mitigation until a developer confirms this is either a bug or outdated documentation. 

**Logs**
<details>
    <summary>Stacktrace from authentik</summary>

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 451, in thread_handler
    raise exc_info[1]
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
    response = await wrapped_callback(
  File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 414, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 408, in wait_for
    return await fut
  File "/usr/local/lib/python3.10/site-packages/asgiref/current_thread_executor.py", line 22, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/asgiref/sync.py", line 455, in thread_handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/views/generic/base.py", line 84, in view
    return self.dispatch(request, *args, **kwargs)
  File "/authentik/policies/views.py", line 58, in dispatch
    self.pre_permission_check()
  File "/authentik/providers/oauth2/views/authorize.py", line 264, in pre_permission_check
    self.params = OAuthAuthorizationParams.from_request(self.request)
  File "/authentik/providers/oauth2/views/authorize.py", line 147, in from_request
    return OAuthAuthorizationParams(
  File "<string>", line 17, in __init__
  File "/authentik/providers/oauth2/views/authorize.py", line 169, in __post_init__
    self.check_redirect_uri()
  File "/authentik/providers/oauth2/views/authorize.py", line 187, in check_redirect_uri
    if not any(fullmatch(x, self.redirect_uri) for x in allowed_redirect_urls):
  File "/authentik/providers/oauth2/views/authorize.py", line 187, in <genexpr>
    if not any(fullmatch(x, self.redirect_uri) for x in allowed_redirect_urls):
  File "/usr/local/lib/python3.10/re.py", line 195, in fullmatch
    return _compile(pattern, flags).fullmatch(string)
  File "/usr/local/lib/python3.10/re.py", line 303, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/local/lib/python3.10/sre_compile.py", line 764, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/local/lib/python3.10/sre_parse.py", line 950, in parse
    p = _parse_sub(source, state, flags & SRE_FLAG_VERBOSE, 0)
  File "/usr/local/lib/python3.10/sre_parse.py", line 443, in _parse_sub
    itemsappend(_parse(source, state, verbose, nested + 1,
  File "/usr/local/lib/python3.10/sre_parse.py", line 668, in _parse
    raise source.error("nothing to repeat",
nothing to repeat at position 0
```
</details>

**Version and Deployment (please complete the following information):**
- authentik version: 2022.5.2
- Deployment: Docker-compose (Managed by Portainer)

**Additional context**
I'm using an app called "step-ca" (Step-CA)
The Step-CA client changes the port used when setting up a local web server to catch the token to forward to the Step-CA server. The wildcard definition is needed in this case to work around this.

**Comment by BeryJu** [comment]
Hi, this is indeed an error in the description of the field, both the description and the error will be fixed in 5.3

**Comment by Hacksawfred3232** [comment]
> Hi, this is indeed an error in the description of the field, both the description and the error will be fixed in 5.3

Okay, thanks for the info. 👍

**Comment by BeryJu** [comment]
5.3 is out so I'll close this
